src/PKSim.Core/Services/ParameterContainerTask.cs

```diff
@@ -67,24 +67,21 @@ IIndividualParametersSameFormulaOrValueForAllSpeciesRepository sameFormulaOrValu
          bool addParameter(ParameterMetaData parameterMetaData)
          {
             //all non individual parameters are added to the spatial structure by default
-            //otherwise, we add all parameters that are used by all species. The value used will be set based on the fact that the parameter has a shared value for all parameters or species or not
-            return parameterMetaData.BuildingBlockType != PKSimBuildingBlockType.Individual || _individualParameterBySpeciesRepository.UsedForAllSpecies(parameterMetaData);
-         }
-         
-         void parameterValueModifier(ParameterMetaData parameterMetaData, IParameter parameter)
-         {
-            //here we know that the parameter fulfills the criteria above but we really only care about parameters that are individual parameters
             if (parameterMetaData.BuildingBlockType != PKSimBuildingBlockType.Individual)
-               return;
+               return true;
+
+            //parameter is not used for all species=> it should not be added to the spatial structure as it will be added dynamically from the individual selected
+            var isUsedForAllSpecies = _individualParameterBySpeciesRepository.UsedForAllSpecies(parameterMetaData);
```

Comment:
@Yuri05 @PavelBal 
I changed the logic when deciding to add a parameter. Before it was added if it was used by ALL species and individual (or not individual parameteR)
Now it's added if it's used by all species AND has the same value or formula
That way it removes the NaN completely that were set explicitly

This seems to work. Since this is really infrastructure deep, if all itnegrations test are passing, we should be good